interface/src/app/app.js

```diff
@@ -19,15 +20,37 @@ angular.module( 'adage', [
   $resourceProvider.defaults.stripTrailingSlashes = false;
 }])
 
-.run( function run () {
+.run( function run ( $state, $location ) {
+
+    // Will only redirect if there is no state AND the url is not a
+    // route for tribe_client
+    String.prototype.startsWith = function(prefix) {
```

Comment:
Nice catch! Changing object prototypes is hazardous for exactly this reason. I'm okay with doing it where necessary, but that should be very rare and it should be done in a way that does not alter the behavior of existing code (including libraries). At minimum we should have tests designed to fail when this type of modification creates a conflict.